app/src/androidTest/java/com/example/triptracker/screens/home/HomeViewScreen.kt

```diff
@@ -0,0 +1,16 @@
+package com.example.triptracker.screens.home
+
+import androidx.compose.ui.test.SemanticsNodeInteractionsProvider
+import io.github.kakaocup.compose.node.element.ComposeScreen
+import io.github.kakaocup.compose.node.element.KNode
+
+class HomeViewScreen(semanticsProvider: SemanticsNodeInteractionsProvider) :
+    ComposeScreen<HomeViewScreen>(
+        semanticsProvider = semanticsProvider, viewBuilderAction = { hasTestTag("HomeScreen") }) {
+
+  // Structural elements of the UI
+  val searchBar: KNode = child { hasTestTag("searchBar") }
+  val itinerarySearch: KNode = searchBar.child { hasSetTextAction() }
+  val itinerary: KNode = child { hasTestTag("Itinerary") }
+  val profilePic: KNode = child { hasTestTag("ProfilePic") }
```

Comment:
This is a good way to structure the screen and this could avoid the use of onNodeWithTag. I don't know which is the preffered way but i think this should be investigated

Reply:
I don't know either, probably for the big components displayed it's good to put it in a separate file like so but for inner components like text in search bar etc I preferred using the onNodeWithTag or with text because I could specify `useUnmergedTree` which made the test find the component.